components/mailer.rst

```diff
@@ -16,8 +16,129 @@ Installation
 
 .. include:: /components/require_autoload.rst.inc
 
+
+Introduction
+------------
+
+
 Usage
 -----
 
-We're currently working on the documentation of this component that was just
-added to Symfony. We'll publish it in a few days.
+    use Symfony\Component\Mailer\Mailer;
+    use Symfony\Component\Mailer\Transport\Smtp\SmtpTransport;
+
+    $transport = new SmtpTransport('localhost');
+    $mailer = new Mailer($transport);
+    $mailer->send($email);
+
+Refer :doc:`Mime component </components/mime>` how to create `$email` object.
+
+
+Transport
+---------
+
+By default, the only transport available in the mailer component is Smtp.
+Below is the list of other popular providers with built in support.
+- Amazon SES  : symfony/amazon-mailer
+- Google Gmail : symfony/google-mailer
+- Mandrill : symfony/mailchimp-mailer
+- Mailgun : symfony/mailgun-mailer
+- Postmark : symfony/postmark-mailer
+- Sendgrid : symfony/sendgrid-mailer
+
+For example to use google's gmail as a transport you need to install symfony/google-mailer.
+
+.. code-block:: terminal
+
+    $ composer require symfony/google-mailer
+
+.. include:: /components/require_autoload.rst.inc
+
+
+    use Symfony\Component\Mailer\Bridge\Google\Smtp\GmailTransport;
+
+    $transport = new GmailTransport('user', 'pass');
+    $transport->send($email);
+
+Use a Dsn
+---------
+
+The mailer component provides a convenient way to create transport object from dsn string.
+
+    use Symfony\Component\Mailer\Transport;
+
+    $transport = Transport::fromDsn($dsn);
+
+Where `$dns` as one of the form below.
+- smtp://user:pass@gmail
+- smtp://key@sendgrid
+- smtp://null
+- smtp://user:pass@mailgun
+- http://key:domain@mailgun
+- api://id@postmark
+
+This provides a unified behaviour across all providers.
+Easily switch from SMTP in dev to a "real" provider in production with same API.
+
+Failover transport
+------------------
+
+You can create failover transport with the help of `||` operator.
+
+Eg : 
+
+    $dsn = 'api://id@postmark || smtp://key@sendgrid';
+
+So if it fails at one transport, the mailer will attempt to send through the other transport.
```

Comment:
```suggestion
So if the first transport fails, the mailer will attempt to send through the second transport.
```

Reply:
Did you missed the `::` . I am not very familiar with rst, so not sure about the purpose.